### PR TITLE
Update-compiler

### PR DESCRIPTION
**Pull request by okikio**
Update the astro compiler package as well as other packages

### CODE REVIEW

File: src/editor/plugins/http.ts [hunk 1]
```diff
@@ -1,12 +1,22 @@
 // Based on https://github.com/okikio/bundle/blob/main/src/ts/plugins/http.ts
 import type { Plugin } from 'esbuild';
 
+export const CACHE = new Map();
```
Comment (natemoo-re) [inline src/editor/plugins/http.ts:4]: Nice, this is very helpful!

File: src/editor/workers/build.ts [hunk 2]
```diff
@@ -56,20 +58,50 @@ let highlighter: Highlighter;
...
+
 const start = (port) => {
     const BuildEvents = new EventEmitter();
-    vol.fromJSON({}, `/`);
+    vol.fromJSON({}, `/`);  
+
+    const postMessage = (obj: any) => {
+        let messageStr = JSON.stringify(obj);
+        let encodedMessage = encode(messageStr);
+        port.postMessage(encodedMessage , [encodedMessage.buffer]);
```
Comment (natemoo-re) [inline src/editor/workers/build.ts:92]: This makes a lot of sense, nice change!

File: src/editor/workers/build.ts [hunk 3]
```diff
@@ -96,31 +128,38 @@ const start = (port) => {
...
         let files = [];
         let html, js, shiki;
 
         if (models.length <= 0 && current) models = [current];
+
         (async () => {
-            try {
-                /* Esbuild */
+            try { 
+                // Preload all components and files to avoid esbuild building while still missing files
```
Comment (natemoo-re) [inline src/editor/workers/build.ts:139]: Thanks for clarifying!

File: src/@astro/internal/index.ts [hunk 4]
```diff
@@ -167,90 +274,131 @@ setup("${astroId}", {${metadata.hydrateArgs ? `value: ${JSON.stringify(metadata.
...
 
-export const renderComponent = async (result: any, displayName: string, Component: unknown, _props: Record<string | number, any>, children: any) => {
+export async function renderComponent(result: SSRResult, displayName: string, Component: unknown, _props: Record<string | number, any>, slots: any = {}) {
   Component = await Component;
-  // children = await renderGenerator(children);
+  const children = await renderSlot(result, slots?.default);
+  
+  // No renderers for the repl
+  // const { renderers } = result._metadata;
+  const renderers = [];
```
Comment (natemoo-re) [inline src/@astro/internal/index.ts:290]: Nice! Was specifically reviewing to see if you handled this.

**Review by natemoo-re — APPROVED** [review]
Looks amazing, thank you SO much! Approved!

**Comment by okikio** [comment]
@natemoo-re This pr is a go for merge

**Pull request merged**